views/js/runner/runner.js

```diff
@@ -0,0 +1,277 @@
+                return this;
+            },
+            
+            /**
+             *
+             * @returns {runner}
+             */
+            previous : function previous(){
+                this.trigger('move', 'previous');
+                return this;
+            },
+            
+            /**
+             *
+             * @returns {runner}
+             */
+            complete : function complete(){
+                this.trigger('complete');
+                return this;
+            },
+            
+            /**
+             *
+             * @param scope
+             * @returns {runner}
+             */
+            exit : function exit(scope){
+                this.trigger('exit', scope);
+                return this;
+            },
+            
+            /**
+             *
+             * @returns {runner}
+             */
+            skip : function skip(){
+                this.trigger('move', 'skip');
+                return this;
+            },
+            
+            /**
+             *
+             * @param itemId
+             * @returns {runner}
+             */
+            jump : function jump(itemId){
+                this.trigger('move', 'jump', itemId);
+                return this;
+            },
+            
+            /**
+             * Set the current state object
+             * @param {Object} state
+             * @returns {runner}
+             */
+            setState : function(state){
+                _state = state;
+                return this;
+            },
+            
+            /**
+             * Return the current state object
+             * @returns {Object}
+             */
+            getState : function(){
+                return _state;
+            },
+            
+            /**
+             * Render the content of the test given the current test state
+             * @returns {runner}
+             */
+            renderContent : function renderContent(){
+                delegate('renderContent', [$contentContainer, _state]);
+                return this;
+            },
+            
+            /**
+             * Infor that the content is rendered and ready for user interaction
+             * @returns {runner}
+             */
+            contentReady : function contentReady(){
+                this.trigger('contentready', $contentContainer);
+                return this;
+            }
+            
+        }).on('move', function move(type, otherArgs_){
+            this.trigger.apply(this, [].slice.call(arguments));
+        });
+        
+        var trigger = runner.trigger;//get the trigger function to overwrite it later
+        runner.trigger = function superTrigger(){
+            var args = [].slice.call(arguments);
+            //implementation note : trigger is a delegated function so the applied context does not matter
+            _logEventTrigger(args);
+            trigger.apply(null, args);
+        };
+            
```

Comment:
Right, I won't disagree with that :)
As @krampstudio , it might be externalised and be an extension of the eventifier instead and I believe that makes sense.